pkg/controllermanager/deployer/deployer.go

```diff
@@ -386,15 +386,19 @@ func (deployer *Deployer) populateBasesAndLocalizations(sub *appsapi.Subscriptio
 			baseTemplate.Labels[known.ClusterNameLabel] = ns.Labels[known.ClusterNameLabel]
 		}
 
-		basesToBeDeleted.Delete(klog.KObj(baseTemplate).String())
-		base, err := deployer.syncBase(sub, baseTemplate)
+		newBaseName := fmt.Sprintf("%s-%s", sub.Namespace, sub.Name)
+		base, err := deployer.syncBase(sub, baseTemplate, newBaseName)
 		if err != nil {
 			allErrs = append(allErrs, err)
 			msg := fmt.Sprintf("Failed to sync Base %s: %v", klog.KObj(base), err)
 			klog.ErrorDepth(5, msg)
 			deployer.recorder.Event(sub, corev1.EventTypeWarning, "FailedSyncingBase", msg)
+			basesToBeDeleted.Delete(klog.KObj(base).String())
```

Comment:
Here `base` will be nil. This will make the old-formatted base be cleaned up later.